diff --git a/node_modules/react-native-track-player/ios/RNTrackPlayer/RNTrackPlayer.swift b/node_modules/react-native-track-player/ios/RNTrackPlayer/RNTrackPlayer.swift
index 64fcbf3..966d46e 100644
--- a/node_modules/react-native-track-player/ios/RNTrackPlayer/RNTrackPlayer.swift
+++ b/node_modules/react-native-track-player/ios/RNTrackPlayer/RNTrackPlayer.swift
@@ -289,6 +289,7 @@ public class RNTrackPlayer: RCTEventEmitter {
             capabilitiesStr.append("togglePlayPause");
         }
         let capabilities = capabilitiesStr.compactMap { Capability(rawValue: $0) }
+        let isLiveStreamSource: Bool = options["isLiveStreamSource"] as? Bool ?? false
 
         let remoteCommands = capabilities.map { capability in
             capability.mapToPlayerCommand(forwardJumpInterval: options["forwardJumpInterval"] as? NSNumber,
@@ -298,6 +299,7 @@ public class RNTrackPlayer: RCTEventEmitter {
                                           bookmarkOptions: options["bookmarkOptions"] as? [String: Any])
         }
 
+        player.nowPlayingInfoController.set(keyValues: [NowPlayingInfoProperty.isLiveStream(isLiveStreamSource)])
         player.enableRemoteCommands(remoteCommands)
 
         resolve(NSNull())
